ecosystem/typescript/sdk/examples/typescript-esm/offer_capabilities.ts

```diff
@@ -0,0 +1,153 @@
+import { AptosAccount, FaucetClient, Network, Provider, HexString, TxnBuilderTypes, BCS, Types } from "aptos";
+import assert from "assert";
+
+const CORE_CODE_ADDRESS = new HexString("0x0000000000000000000000000000000000000000000000000000000000000001");
+const ED25519_ACCOUNT_SCHEME = 0;
+
+// Works for both SignerCapabilityOffer and RotationCapabilityOffer
+// Except the structName fields are different and the chainId isn't used in SignerCapabilityOffer
+type CapabilityOfferProofChallengeV2 = {
+  accountAddress: TxnBuilderTypes.AccountAddress;
+  moduleName: string;
+  structName: string;
+  sequenceNumber: number;
+  sourceAddress: TxnBuilderTypes.AccountAddress;
+  recipientAddress: TxnBuilderTypes.AccountAddress;
+  chainId?: number; // not used in SignerCapabilityOffer
```

Comment:
From a BCS approach, it would be better to Make RotationCapabilityOffer contain a SignerCapabilityOffer.

I would further suggest though that we actually inline the first three fields, that shouldn't be part of the input